pkg/kubectl/cmd/apply.go

```diff
@@ -245,7 +245,13 @@ func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 	// Create the versioned struct from the type defined in the restmapping
 	// (which is the API version we'll be submitting the patch to)
 	versionedObject, err := api.Scheme.New(p.mapping.GroupVersionKind)
-	if err != nil {
+	switch {
+	case runtime.IsNotRegisteredError(err):
+		versionedObject, _, err = p.decoder.Decode(current, nil, nil)
```

Comment:
> Do you mean JSONPatchType or MergePatchType?

I'm not sure. The first step would be computing a patch in the absence of a go struct giving schema info.

> I guess you mean option 2 in kubernetes/community#229 (comment)

no, that's referring to specific behavior in union type fields. I'm talking about what the client behavior should be when it has *no* schema info for a particular type, just before and after JSON serializations it is trying to apply